platform/darwin/src/NSPredicate+MGLAdditions.mm

```diff
@@ -294,13 +294,56 @@ + (instancetype)mgl_predicateWithJSONObject:(id)object {
         return [NSPredicate predicateWithValue:YES];
     }
     if ([op isEqualToString:@"all"]) {
-        NSArray *subpredicates = MGLSubpredicatesWithJSONObjects([objects subarrayWithRange:NSMakeRange(1, objects.count - 1)]);
+        NSArray *jsonObjects = [objects subarrayWithRange:NSMakeRange(1, objects.count - 1)];
+        NSArray<NSPredicate *> *subpredicates = MGLSubpredicatesWithJSONObjects(jsonObjects);
+        if (jsonObjects.count == 2) {
```

Comment:
From this line onward, there should be no references to `jsonObjects`. Use `subpredicates` instead.

/ref https://github.com/mapbox/mapbox-gl-native/pull/11632#discussion_r180969287